lib/host-sp-messages/src/lib.rs

```diff
@@ -175,6 +187,183 @@ pub enum KeyLookupResult {
     MaxResponseLenTooShort,
 }
 
+/// Results for an inventory data request
+///
+/// These **cannot be reordered**; the host and SP must agree on them.
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize, SerializedSize,
+)]
+pub enum InventoryDataResult {
+    Ok,
+    /// The given index is larger than our device count
+    InvalidIndex,
+    /// Communication with the device failed in a way that suggests its absence
+    DeviceAbsent,
+    /// Communication with the device failed in some other way
+    DeviceFailed,
+    /// Failed to serialize data
+    SerializationError,
+}
+
+impl From<HubpackError> for InventoryDataResult {
+    fn from(_: HubpackError) -> Self {
+        Self::SerializationError
+    }
+}
+
+impl From<drv_i2c_api::ResponseCode> for InventoryDataResult {
+    fn from(e: drv_i2c_api::ResponseCode) -> Self {
+        match e {
+            drv_i2c_api::ResponseCode::NoDevice => {
+                InventoryDataResult::DeviceAbsent
+            }
+            _ => InventoryDataResult::DeviceFailed,
+        }
+    }
+}
+
+/// Data payload for an inventory data request
+///
+/// These **cannot be reordered**; the host and SP must agree on them.  New
+/// variants may be added to the end, and existing variants may be extended with
+/// new data (at the end), but no changes should be made to existing bytes.
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize, SerializedSize,
+)]
+pub enum InventoryData {
+    /// Raw DIMM data
+    #[serde(with = "BigArray")]
+    DimmSpd([u8; 512]),
+
+    /// Device or board identity, typically stored in a VPD EEPROM
+    VpdIdentity(oxide_barcode::VpdIdentity),
```

Comment:
Just to make sure we're on the same page here, if for some reason the sizing of the VPD Identity were ever to change beyond the 11/11 characters for the MPN/SN, we would introduce a newer type here. This does make me wonder if we should be passing something here where we use either a larger size than the others and zero-terminate this or have an explicit integer encoding lengths so we're ok to change this in the future.

In particular, the existing `MODEL_LEN` and `SERIAL_LEN` which are used by the `Identity` structure are 51 characters (50 + `\0`), so given that it is already future proofed, I wonder if this should be consistent or not.

Reply:
It's correct that moving away from `oxide_barcode::VpdIdentity` would require making a new message type for `InventoryData::VpdIdentity` and `InventoryData::FanIdentity`.

Do you think it would make sense to use `host_sp_messages::Identity` (which stores the 51-byte values) instead here?  It's hard to imagine that encoding becoming defunct.